### DISCUSSION TITLE
build

### DISCUSSION THREAD

**Discussion by Cleydsonsl** [callstack/super-app-showcase#35] [General]
Hello, I wanted to know how to generate a build, I will try to explain.

I'm using your tech to build a superapp with mini-apps building it in general, the miniapps I'm doing in separate repo because each squad will act in its own repo but I'm having difficulties trying to simulate a productive environment with docker uploading the build generated by bundle-ios and bundle-android the host app locally cannot see the miniapps in docker, the objective is to publish only the superapp in the stores making it observe only the miniapps builds in some hosting.

**Comment by thymikee** [comment]
Hey! Do you have your bundles deployed in a readable location? To make sure that the host app can actually download the bundles

**Comment by jbroma** [comment]
Could you please describe your setup in more detail? Personally I still cant figure out how Docker fits into the whole setup in here.

**Comment by Cleydsonsl** [comment]
here is my repo: https://github.com/Cleydsonsl/superapp

if it is not feasible please help me to see a possible way out, we would like each team to be able to work on their module without having to use the host.

**Reply by Cleydsonsl** [reply]
@jbroma  @thymikee

**Comment by eeuruetade** [comment]
I did something similar, my dockerfile looks like this

```
FROM nginx:alpine
WORKDIR /usr/share/nginx/html

COPY build /usr/share/nginx/html
COPY nginx.conf /etc/nginx/nginx.conf

EXPOSE 80

CMD ["nginx", "-g", "daemon off;"]
```

My nginx.config

```
events {
    worker_connections 1024;
}

http {
    server {
        listen 80;
        server_name localhost;

        location / {
            root /usr/share/nginx/html;
            autoindex on;
            try_files $uri $uri/ =404;
        }
    }
}
```

The ports you expose must be enabled for the app to take them correctly

`adb reverse tcp:8081 tcp:8081 && adb reverse tcp:9000 tcp:9000`

To finish in my index.js file

```
const containers = {
  app1: `http://localhost:9000/outputs/${Platform.OS}/remotes/[name][ext]`,
  //app1: 'http://localhost:9000/[name][ext]',
};

const resolveURL = Federated.createURLResolver({
  containers,
});
```
